### PR TITLE
Oop design principles homework

### PR DESCRIPTION
**Pull request by Daniel-Cretu**

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -17,8 +19,11 @@ class AppServiceProvider extends ServiceProvider
     public function register()
     {
         $this->app->bind(RequestActivityLoggerInterface::class, function () {
-            return $this->app->make(DummyRequestActivityLogger::class);
-            // Environment type. Return sau Debug sau Production
+            if(strtolower(env('APP_ENV')) === 'production')
```
Comment (egorbwork) [inline app/Providers/AppServiceProvider.php]: Can't accept, the env helper is not available in cache environment, so it is broken.

File: app/Console/Commands/RoulettePlay.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                $this->info("Congrats! Back to zero leader score. Reason: You died !");
+                $statistics = $this->cacheRepository->get('Roulette results', []);
+                $statistics[$name] = 0;
+
+                $this->cacheRepository->set('Roulette results', $statistics, 86400);
+
+                $killstreak = $this->cacheRepository->get('Kill-streak', 0);
+                $killstreak++;
+                $this->cacheRepository->set('Kill-streak', $killstreak, 86400);
```
Comment (egorbwork) [inline app/Console/Commands/RoulettePlay.php:72]: Please keep in statistic, also number of times specific number was chosen. and keep it in statistics command.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Looks not bad, but exists bugs and I would prefer to extend solution.

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (Daniel-Cretu) [reply app/Providers/AppServiceProvider.php]: Now using config() helper instead of env().

File: app/Console/Commands/RoulettePlay.php (hunk 2 above)
Comment (Daniel-Cretu) [reply app/Console/Commands/RoulettePlay.php:72]: Now showing kill-streak on statistics command.
Now keeping in cache Cursed Numbers at which players loose the game. Also showing them in statistics.

**Review by egorbwork — APPROVED** [review]
Now looks complete solution. Good job!

**Pull request merged**